SimPEG/EM/NSEM/RxNSEM.py

```diff
@@ -0,0 +1,696 @@
+    @property
+    def _ahz_py(self):
+        return mkvc(mkvc(self.f[self.src,'b_py'],2).T/mu_0*self.Pbz.T)
+
+    # NOTE: need to add a .T at the end for the output to be (nU,)
+    def _aex_px_u(self, vec):
+        """
+        """
+        # vec is (nD,) and returns a (nU,)
+        return self.f._e_pxDeriv_u(self.src,self.Pex.T*mkvc(vec,),adjoint=True)
+    def _aey_px_u(self, vec):
+        """
+        """
+        # vec is (nD,) and returns a (nU,)
+        return self.f._e_pxDeriv_u(self.src,self.Pey.T*mkvc(vec,),adjoint=True)
+    def _aex_py_u(self, vec):
+        """
+        """
+        # vec is (nD,) and returns a (nU,)
+        return self.f._e_pyDeriv_u(self.src,self.Pex.T*mkvc(vec,),adjoint=True)
+    def _aey_py_u(self, vec):
+        """
+        """
+        # vec is (nD,) and returns a (nU,)
+        return self.f._e_pyDeriv_u(self.src,self.Pey.T*mkvc(vec,),adjoint=True)
+    def _ahx_px_u(self, vec):
+        """
+        """
+        # vec is (nD,) and returns a (nU,)
+        return self.f._b_pxDeriv_u(self.src,self.Pbx.T*mkvc(vec,),adjoint=True)/mu_0
+    def _ahy_px_u(self, vec):
+        """
+        """
+        # vec is (nD,) and returns a (nU,)
+        return self.f._b_pxDeriv_u(self.src,self.Pby.T*mkvc(vec,),adjoint=True)/mu_0
+    def _ahz_px_u(self, vec):
+        """
+        """
+        # vec is (nD,) and returns a (nU,)
+        return self.f._b_pxDeriv_u(self.src,self.Pbz.T*mkvc(vec,),adjoint=True)/mu_0
+    def _ahx_py_u(self, vec):
+        """
+        """
+        # vec is (nD,) and returns a (nU,)
+        return self.f._b_pyDeriv_u(self.src,self.Pbx.T*mkvc(vec,),adjoint=True)/mu_0
+    def _ahy_py_u(self, vec):
+        """
+        """
+        # vec is (nD,) and returns a (nU,)
+        return self.f._b_pyDeriv_u(self.src,self.Pby.T*mkvc(vec,),adjoint=True)/mu_0
+    def _ahz_py_u(self, vec):
+        """
+        """
+        # vec is (nD,) and returns a (nU,)
+        return self.f._b_pyDeriv_u(self.src,self.Pbz.T*mkvc(vec,),adjoint=True)/mu_0
+    # Define the components of the derivative
+    @property
+    def _aHd(self):
+        return self._sDiag(1./(
+            self._sDiag(self._ahx_px)*self._ahy_py -
+            self._sDiag(self._ahx_py)*self._ahy_px
+        ))
+    def _aHd_uV(self, x):
+        return (self._ahx_px_u(self._sDiag(self._ahy_py)*x) +
+            self._ahx_px_u(self._sDiag(self._ahy_py)*x) -
+            self._ahy_px_u(self._sDiag(self._ahx_py)*x) -
+            self._ahx_py_u(self._sDiag(self._ahy_px)*x)
+        )
+
+    def eval(self):
+        """
+        Function to evaluate datum for this receiver
+        """
+        raise NotImplementedError('SimPEG.EM.NSEM receiver has to have an eval method')
+
+    def evalDeriv(self):
+        """
+        Function to evaluate datum for this receiver
+        """
+        raise NotImplementedError('SimPEG.EM.NSEM receiver has to have an evalDeriv method')
+
+
+class rx_Point_impedance1D(SimPEG.Survey.BaseRx):
```

Comment:
Doesn't that counter dict then namespace convention, since we have multiple `Fields...` and `Problem...` in the base name space.? 
Should we then just access the receivers via  `NSEM.RxNSEM.Point_impedance1D`? 

I  fine with all those ways.